pilot/util/processes.py

```diff
@@ -567,3 +568,11 @@ def cleanup(job):
 
     logger.info("will now attempt to kill all subprocesses of pid=%d" % job.pid)
     kill_processes(job.pid)
+
+
+def get_thread_tid():
+    """System call gettid on Linux, returning thread-id."""
+    try:
+        return ctypes.CDLL('libc.so.6').syscall(186)
```

Comment:
Personally I don't like low level calls like that, but I find no alternative that works properly.. Do we need this in the longer run? The function name should be get_thread_id() and not get_thread_tid() but it's irrelevant if we only need this for a short while.

Reply:
in python, there is another thread id which is Thread.ident. it's a long number. Tid is from linux os. In linux os, this one is called tid. I put it here just thinking maybe in the future we need it for debug. I will remove it.